app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,69 @@
+samson.controller('ProjectRolesCtrl', function($scope, $element, $filter, userProjectRoleFactory, projectRoleFactory, projectRolesService, messageCenterService) {
+  $scope.project_role = {};
+  $scope.roles = [];
+
+  $scope.initModel = function() {
+    $scope.project_role = userProjectRoleFactory.buildFromDom($element[0]);
+    loadProjectRoles();
+  };
+
+  $scope.$watch('project_role.role_id', function(new_role_value, old_role_value) {
```

Comment:
Watches can be a bit hairy in angular and can add a bit too much overhead to a page. I'm possibly a bit out of date on angular, but would a ng-change event bind on the control not be better?
```
<input type="radio" ng-model="project_role.role_id" ng-value="role.id" ng-change="roleChanged()" />
```